examples/nlp/token_classification/punctuate_capitalize_infer.py

```diff
@@ -0,0 +1,739 @@
+    )
+    parser.add_argument(
+        "--margin",
+        "-g",
+        type=int,
+        default=16,
+        help="A number of subtokens in the beginning and the end of segments which output probabilities are not used "
+        "for prediction computation. The first segment does not have left margin and the last segment does not have "
+        "right margin. For example, if input sequence is tokenized into characters, `max_seq_length=5`, `step=1`, "
+        "and `margin=1`, then query 'hello' will be tokenized into segments `[['[CLS]', 'h', 'e', 'l', '[SEP]'], "
+        "['[CLS]', 'e', 'l', 'l', '[SEP]'], ['[CLS]', 'l', 'l', 'o', '[SEP]']]`. These segments are passed to the "
+        "model. Before final predictions computation, margins are removed. In the next list, subtokens which logits "
+        "are not used for final predictions computation are marked with asterisk: `[['[CLS]'*, 'h', 'e', 'l'*, "
+        "'[SEP]'*], ['[CLS]'*, 'e'*, 'l', 'l'*, '[SEP]'*], ['[CLS]'*, 'l'*, 'l', 'o', '[SEP]'*]]`.",
+    )
+    parser.add_argument(
+        "--batch_size", "-b", type=int, default=128, help="Number of segments which are processed simultaneously.",
+    )
+    args = parser.parse_args()
+    if args.input_manifest is None and args.output_manifest is not None:
+        parser.error("--output_manifest requires --input_manifest")
+    if args.pretrained_name is None and args.model_path is None:
+        setattr(args, default_model_parameter, default_model)
+    for name in ["input_manifest", "input_text", "output_manifest", "output_text", "model_path"]:
+        if getattr(args, name) is not None:
+            setattr(args, name, getattr(args, name).expanduser())
+    return args
+
+
+def load_manifest(manifest: Path):
+    result = []
+    with manifest.open() as f:
+        for i, line in enumerate(f):
+            data = json.loads(line)
+            result.append(data)
+    return result
+
+
+def _get_subtokens_and_subtokens_mask(query: str, tokenizer: TokenizerSpec) -> Tuple[List[str], List[int]]:
+    """
+    Tokenizes input query into subtokens and creates subtokens mask. Subtokens mask is an array of the same length as
+    subtokens array and contains zeros and ones in which. If element of mask equals 1, then corresponding subtoken in
+    subtokens array is first subtoken in some word
+    Args:
+        query: a string that will be tokenized
+        tokenizer: an instance of tokenizer
+    Returns:
+        subtokens: list of subtokens
+        subtokens_mask: list of ints
+    """
+    words = query.strip().split()
+    subtokens = []
+    subtokens_mask = []
+    for j, word in enumerate(words):
+        word_tokens = tokenizer.text_to_tokens(word)
+        subtokens.extend(word_tokens)
+        subtokens_mask.append(1)
+        subtokens_mask.extend([0] * (len(word_tokens) - 1))
+    return subtokens, subtokens_mask
+
+
+def _check_max_seq_length_and_margin_and_step(max_seq_length: int, margin: int, step: int):
```

Comment:
move to the model class or create a utils file inside https://github.com/NVIDIA/NeMo/tree/main/nemo/collections/nlp/models/token_classification

Reply:
Moved to dataset file